client/memory.c

```diff
@@ -131,6 +134,10 @@ int xdag_mem_init(size_t size)
 void *xdag_malloc(size_t size)
 {
 	uint8_t *res;
+	
+	if (size <= 0) {
```

Comment:
size_t is unsigned, and why are you moving it here?
It is very unlikely that someone will ask to malloc 0

Reply:
the function was designed for other one to use, maybe for false usage, size to be zero, not by intension, so I think this check is necessary, in fact I just move this check to the first line

Reply:
size_t is unsigned, and why are you moving it here?
It is very unlikely that someone will ask to malloc 0

Reply:
maybe u have noticed I didn't realize size_t is unsigned, its my negligence， it should be
if (size) return 0;
I didn't read code of malloc, so I didn't know malloc(0)'s return value
for safe reason, it's true to check before it 
in fact , the following is man of malloc
```
The malloc() function allocates size bytes and returns a pointer to the allocated memory. The memory is not initialized. If size is 0, then malloc() returns either NULL, or a unique pointer value that can later be successfully passed to free().
```

Reply:
in fact, I just realize u use malloc(0), I think this can answer your question: who  use size = 0 :p